modules/standard/Path.chpl

```diff
@@ -189,4 +189,25 @@ proc file.realPath(): string {
    }
  }
 
+ /*
+
+ Returns the parent directory of the :type:`~IO.file` record.  For instance,
+ a file with path `/foo/bar/baz` would return `/foo/bar`
+
+  :return: The parent directory of the file
+  :rtype: `string`
+
+ */
+ proc file.getParentName(out error:syserr): string {
```

Comment:
Oh shoot, I forgot:  we've been having these functions come in pairs - a version that provides the syserr and a version that just halts on failure with a reasonable error message.  The set up is a little nitpicky, so I can do it after the merge unless you want to own all the changes to this new function.  Your call :)